base/c.jl

```diff
@@ -548,6 +548,11 @@ function expand_ccallable(rt, def)
     error("expected method definition in @ccallable")
 end
 
+"""
+    @ccallable(def)
+
+Julia app with `ccallable` entry-point means that the respective Julia function can be called directly from C
```

Comment:
What is "Julia app"?

Reply:
Yeah, this description is a bit too PackageCompiler specific. I would suggest something like:

> `@ccallable`
>
> Make the annotated function be callable from C using its name. This can, for example, be used to expose functionality as a C-API when creating a custom Julia sysimage.

Or something along those lines...